ktor-server/ktor-server-host-common/nix/src/io/ktor/server/engine/ShutdownHookNative.kt

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.engine
+
+import kotlinx.cinterop.*
+import platform.posix.*
+
+/**
+ * Adds automatic application shutdown hooks management. Should be used **before** starting the engine.
+ * Once application termination noticed, [stop] block will be executed.
+ * Please note that a shutdown hook only registered when the application is running. If the application
+ * is already stopped then there will be no hook and no [stop] function invocation possible.
+ * So [stop] block will be called once or never.
+ */
+public actual fun ApplicationEngine.addShutdownHook(stop: () -> Unit) {
+//    signal(SIGINT, staticCFunction<Int, Unit> {
```

Comment:
uncomment?

Reply:
Nope, it doesn't work because capturing is prohibited here. Kotlin native proposed to rewrite using a global variable. Will fix.